docs/src/dev-docs/design-deployment-orchestration.md

```diff
@@ -0,0 +1,241 @@
+# Deployment orchestration
+
+The CLP package is composed of several components that are currently designed to be deployed in a
+set of containers that are orchestrated using a framework like [Docker Compose][docker-compose].
+This document explains the architecture of that orchestration and any associated nuances.
+
+## Architecture
+
+[Figure 1](#figure-1) shows the components (*services* in orchestrator terminology) in the CLP
+package as well as their dependencies. The CLP package consists of several long-running services
+(e.g., `database`) and some one-time initialization jobs (e.g., `db-table-creator`). Some of the
+long-running services depend on the successful completion of the one-time jobs (e.g., `webui`
+depends on `results-cache-indices-creator`), while others depend on the health of other long-running
+services (e.g., `compression-scheduler` depends on `queue`).
+
+[Table 1](#table-1) below lists the services their functions, while [Table 2](#table-2) lists the
+one-time initialization jobs and their functions.
+
+(figure-1)=
+::::{card}
+
+:::{mermaid}
+%%{
+    init: {
+        "theme": "base",
+        "themeVariables": {
+            "primaryColor": "#0066cc",
+            "primaryTextColor": "#fff",
+            "primaryBorderColor": "transparent",
+            "lineColor": "#007fff",
+            "secondaryColor": "#007fff",
+            "tertiaryColor": "#fff"
+        }
+    }
+}%%
+graph LR
+  %% Services
+  database["database (MySQL)"]
+  queue["queue (RabbitMQ)"]
+  redis["redis (Redis)"]
+  results_cache["results-cache (MongoDB)"]
+  compression_scheduler["compression-scheduler"]
+  query_scheduler["query-scheduler"]
+  compression_worker["compression-worker"]
+  query_worker["query-worker"]
+  reducer["reducer"]
+  webui["webui"]
+  garbage_collector["garbage-collector"]
+
+  %% One-time jobs
+  db_table_creator["db-table-creator"]
+  results_cache_indices_creator["results-cache-indices-creator"]
+
+  %% Dependencies
+  database -->|healthy| db_table_creator
+  results_cache -->|healthy| results_cache_indices_creator
+  db_table_creator -->|completed_successfully| compression_scheduler
+  queue -->|healthy| compression_scheduler
+  redis -->|healthy| compression_scheduler
+  db_table_creator -->|completed_successfully| query_scheduler
+  queue -->|healthy| query_scheduler
+  redis -->|healthy| query_scheduler
+  query_scheduler -->|healthy| reducer
+  results_cache_indices_creator -->|completed_successfully| reducer
+  db_table_creator -->|completed_successfully| webui
+  results_cache_indices_creator -->|completed_successfully| webui
+  db_table_creator -->|completed_successfully| garbage_collector
+  results_cache_indices_creator -->|completed_successfully| garbage_collector
+
+  subgraph Databases
+    database
+    queue
```

Comment:
(i understand it was me who initially propose the name. back then i couldn't think of a better name) technically RabbitMQ is more of a message broker than a database? would it be better to name this group as "Data layer" or "Data infra", or those names would confuse people about where our archives are stored? or maybe something like "State Persistence & Messaging"?

Reply:
How about "Shared data stores"?